src/Microsoft.AspNet.Mvc.Core/ActionResults/FilePathResult.cs

```diff
@@ -122,7 +121,12 @@ internal string ResolveFilePath(IFileProvider fileProvider)
             {
                 // The path is relative and IFileProvider found the file, so return the full
                 // path.
-                return fileInfo.PhysicalPath;
+                return new ResolveFilePathResult()
+                {
+                    // Note that physical path could be null in case of non-disk file (ex: embedded resource)
+                    PhysicalFilePath = fileInfo.PhysicalPath,
+                    FileInfo = fileInfo
+                };
             }
```

Comment:
Good catch @rynowak ...I did initially observe this and wanted to make changes where we we can check for the presence of rooted path with something like `File.Exists` from System.IO, but didn't do it due to the following comment:

https://github.com/aspnet/Mvc/blob/issue2086-filepathresult/src/Microsoft.AspNet.Mvc.Core/ActionResults/FilePathResult.cs#L101

Like my question is around if `File.Exists` works across all operating systems...probably should following up regarding this...